socs/db/suprsync.py

```diff
@@ -278,6 +334,116 @@ def get_known_files(self, archive_name, session=None):
         )
 
         return list(query.all())
+    
+    def _add_file_tcdir(self, file: SupRsyncFile, session):
+        """
+        Creates and adds a TimecodeDir for a file if possible.  This will
+        attempt to extract the timecode from the remote filename, and will
+        create a new TimecodeDir if it doesn't already exist.
+        """
+        # print(file.remote_path)
+        tc = check_timecode(file)
+        if tc is None:
+            return None
+
+        tcdir = session.query(TimecodeDir).filter(
+            TimecodeDir.timecode==tc,
+            TimecodeDir.archive_name == file.archive_name,
+        ).one_or_none()
+
+        if tcdir is not None:
+            return tcdir
+
+        tcdir = TimecodeDir(timecode=tc, archive_name=file.archive_name)
+        session.add(tcdir)
+        return tcdir
+    
+    def create_all_timecode_dirs(self, archive_name):
+        with self.Session.begin() as session:
+            files = self.get_known_files(archive_name, session=session)
+            for file in files:
+                self._add_file_tcdir(file, session)
+    
+    def update_all_timecode_dirs(self, archive_name, file_root, sync_id):
+        with self.Session.begin() as session:
+            tcdirs = session.query(TimecodeDir).all()
+            for tcdir in tcdirs:
+                self._update_tcdir(tcdir, session, file_root, sync_id)
+    
+    def _update_tcdir(self, tcdir, session, file_root, sync_id):
+        """
+        Takes the next series of actions for a timecode dir object.
+        - If we expect no more files to be added to the tc dir, marks it as
+          complete
+        - If all files in the tc dir have been synced, marks it as synced
+        - If the tc dir is synced and not finalized, creates the finalization
+          file and marks as finalized.
+        """
+        if tcdir.finalized:
+            return
+        
+        if not tcdir.completed:
+            all_tcs = session.query(TimecodeDir.timecode).all()
+            for tc, in all_tcs:
+                if tc > tcdir.timecode:
+                    # Mark as complete if there's a timecode after this one
+                    tcdir.completed = True
+                    break
+            else:
+                # No timecodes after this one. Mark after complete if we are
+                # over a full day away.
+                if (time.time()//1e5 - tcdir.timecode) > 1:
+                    tcdir.completed = True
```

Comment:
Please make this auto-completion time a constant at the top of the program.

Reply:
This has been resolved but it's not showing it in the textbox, though it's been marked as outdated. If you look at the "files changed" I did indeed replace all the `time.time()` calls with a constant.

Reply:
unless I'm misinterpreting the feedback

Reply:
Talked with brian about this in slack and decided to move the 1 here to a constant defined at the top of the file. I think I misinterpreted @mhasself's initial comment.